openpype/lib/delivery.py

```diff
@@ -4,27 +4,41 @@
 import glob
 import clique
 import collections
+import re
 
 
 def collect_frames(files):
     """
         Returns dict of source path and its frame, if from sequence
 
-        Uses clique as most precise solution
+        Uses clique as most precise solution, used when anatomy template that
+        created files is not known.
+
+        Depends that version substring starts with 'v' with any number of
+        numeric characters after.
 
         Args:
-            files(list): list of source paths
+            files(list) or (set with single value): list of source paths
         Returns:
             (dict): {'/asset/subset_v001.0001.png': '0001', ....}
     """
     collections, remainder = clique.assemble(files, minimum_items=1)
 
+    real_file_name = None
     sources_and_frames = {}
+    if len(files) == 1:
+        real_file_name = list(files)[0]
+        sources_and_frames[real_file_name] = None
+
     if collections:
         for collection in collections:
             src_head = collection.head
             src_tail = collection.tail
 
+            # version recognized as a collection
+            if re.match(".*([a-zA-Z0-9]%[0-9]+d).*", collection.format()):
```

Comment:
Wouldn't you be better of taking just the info from the collection itself instead of trying to parse the formatted string (which I'd consider just to be consistent for user reading purposes instead of parsing purposes). I wouldn't expect the clique api to try and keep the formatting api super consistent across versions since it's just there to format a default readable format? (unless you explicitly pass it the string to format?)

Looking at your logic you're saying that if there's directly any alphabetic character or number before the parsed frame number) then you consider it to be a parsed version?

Somehow this logic doesn't feel intuitive to me. What exactly are we trying to solve?

Reply:
Cleaned up, used `clique.PATTERNS["frames"]` which handles it more nicely